packages/ui/hooks/index.ts

```diff
@@ -153,8 +154,9 @@ export function useExpiringLocalPreference<Data = any>(
             const expiresAt = Number.isInteger(data?.expires)
                ? data.expires
                : 0;
-            if (expiresAt && Date.now() < expiresAt) {
-               setData(data?.value);
+            const validData = validator(data?.value);
+            if (validData !== null && expiresAt && Date.now() < expiresAt) {
+               setData(validData);
```

Comment:
we could treat the validator returning anything besides null as valid and then `setData(data?.value)` so we wouldn't need the ternary operations in the validators anymore

Reply:
I don't really understand, it seems like what we're doing here: 

```
if (validData !== null) { setData(data) }
```

Reply:
sorry what i meant to propose is having the validators return a boolean value so like instead of 
```php
data === ProductViewType.List || data === ProductViewType.Grid
            ? data
            : null
```

we could just have 
```php
data === ProductViewType.List || data === ProductViewType.Grid
```

Reply:
Oh, I forgot that typescript can do assertion functions...,..

Update: I tried but I couldn't get typescript to enforce that the validator function was checking the type. It's easier to enforce that it returns the right type.


For instance, using a validator type of:
`validator: (data: any) => data is Data`
and a validation function of:
`(data: any): data is ProductViewType => data`

Results in typescript not complaining. It apparently doesn't validate that your assertion function actually asserts anything.